version: '3.9'

services:

  db:
    image: postgres:13.0-alpine
    restart: on-failure
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    volumes:
      - pg_data:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}
      interval: 5s
      timeout: 3s
      retries: 3
    networks:
      - backend_nw

  api:
    image: ${DOCKER_HUB_USERNAME}/todolist:${TAG_NAME}
    restart: always
    env_file:
      - ./.env
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      test: python3 -c 'import http.client;http.client.HTTPConnection("127.0.0.1:8000", timeout=1).request("GET", "/ping/")'
      interval: 3s
      timeout: 3s
      retries: 3
    networks:
      - backend_nw
      - frontend_nw

  frontend:
    image: sermalenk/skypro-front:lesson-36
    restart: always
    ports:
      - "80:80"
    depends_on:
      api:
        condition: service_healthy
    networks:
      - frontend_nw
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - ./static/:/opt/static/

volumes:
  pg_data:

networks:
  backend_nw:
    driver: bridge
  frontend_nw:
    driver: bridge
